semantic-conventions/src/opentelemetry/semconv/model/semantic_attribute.py

```diff
@@ -250,25 +250,25 @@ def parse_id(attribute):
             position = attribute.lc.data[list(attribute)[0]]
             msg = "Non array examples for {} are not allowed".format(attr_type)
             raise ValidationError.from_yaml_pos(position, msg)
-        if (
-            (is_simple_type or isinstance(attr_type, EnumAttributeType))
-            and not isinstance(examples, CommentedSeq)
-            and examples is not None
-        ):
+        if not isinstance(examples, CommentedSeq) and examples is not None:
+            # TODO: If validation fails later, this will crash when trying to access positio data
+            # since a list, contratry to a CommentedSeq, does not have position data
             examples = [examples]
-        if is_simple_type and attr_type not in [
+        if is_simple_type and attr_type not in (
             "boolean",
             "boolean[]",
             "int",
             "int[]",
             "double",
             "double[]",
-        ]:
-            if examples is None or (len(examples) == 0):
+        ):
+            if not examples:
                 position = attribute.lc.data[list(attribute)[0]]
                 msg = "Empty examples for {} are not allowed".format(attr_type)
                 raise ValidationError.from_yaml_pos(position, msg)
-        if is_simple_type and attr_type not in ["boolean", "boolean[]"]:
```

Comment:
@thisthat Any idea why this was originally excluded? To support more boolean values like "yes" instead of "true"?

Reply:
I think we excluded boolean attributes from having an example because it is not very useful to have `true`/`false` there.

Reply:
I know they are not required to have examples (though before this patch, non-array bools without examples would have crashed the generator). But in case they do have examples (which is allowed), the examples should still be checked for whether they have the right type.